sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -247,6 +270,140 @@ def test_lasso_cv_positive_constraint():
+    if 'alpha' in params:
+        model_normalize.set_params(alpha=params['alpha'])
+        if model_name in ['Lasso', 'LassoLars', 'MultiTaskLasso']:
+            new_params = dict(
+                alpha=params['alpha'] * np.sqrt(X_train.shape[0]))
+        if model_name in ['Ridge', 'RidgeClassifier']:
+            new_params = dict(alpha=params['alpha'] * X_train.shape[0])
+    if model_name in ['ElasticNet', 'MultiTaskElasticNet']:
+        if params['l1_ratio'] == 1:
+            new_params = dict(
+                alpha=params['alpha'] * np.sqrt(X_train.shape[0]))
+        if params['l1_ratio'] == 0:
+            new_params = dict(alpha=params['alpha'] * X_train.shape[0])
+
+    if 'new_params' in locals():
+        pipeline[1].set_params(**new_params)
+
+    model_normalize.fit(X_train, y_train)
+    y_pred_normalize = model_normalize.predict(X_test)
+
+    pipeline.fit(X_train, y_train)
+    y_pred_standardize = pipeline.predict(X_test)
+
+    assert_allclose(
+        model_normalize.coef_ * pipeline[0].scale_, pipeline[1].coef_)
+    assert pipeline[1].intercept_ == pytest.approx(y_train.mean())
+    assert (model_normalize.intercept_ ==
+            pytest.approx(y_train.mean() -
+                          model_normalize.coef_.dot(X_train.mean(0))))
+    assert_allclose(y_pred_normalize, y_pred_standardize)
+
+
+@pytest.mark.parametrize(
+    "LinearModel, params",
+    [(Lasso, {"tol": 1e-16, "alpha": 0.1}),
+     (LassoCV, {"tol": 1e-16}),
+     (ElasticNetCV, {}),
+     (RidgeClassifier, {"solver": 'sparse_cg', "alpha": 0.1}),
+     (ElasticNet, {"tol": 1e-16, 'l1_ratio': 1, "alpha": 0.01}),
+     (ElasticNet, {"tol": 1e-16, 'l1_ratio': 0, "alpha": 0.01}),
+     (Ridge, {"solver": 'sparse_cg', 'tol': 1e-12, "alpha": 0.1}),
+     (LinearRegression, {}),
+     (RidgeCV, {})]
+ )
+def test_model_pipeline_same_dense_and_sparse(LinearModel, params):
+    # Test that linear model preceeded by StandardScaler in the pipeline and
+    # with normalize set to False gives the same y_pred and the same .coef_
+    # given X sparse or dense
+
+    model_dense = make_pipeline(
+        StandardScaler(with_mean=False),
+        LinearModel(normalize=False, **params)
+    )
+
+    model_sparse = make_pipeline(
+        StandardScaler(with_mean=False),
+        LinearModel(normalize=False, **params)
+    )
+
+    # prepare the data
+    rng = np.random.RandomState(0)
+    n_samples = 200
+    n_features = 2
+    X = rng.randn(n_samples, n_features)
+    X[X < 0.1] = 0.
+
+    X_sparse = sparse.csr_matrix(X)
+    y = rng.rand(n_samples)
+
+    if is_classifier(LinearModel):
+        y = np.sign(y)
+
+    model_dense.fit(X, y)
+    model_sparse.fit(X_sparse, y)
+
+    assert_allclose(model_sparse[1].coef_, model_dense[1].coef_)
+    y_pred_dense = model_dense.predict(X)
+    y_pred_sparse = model_sparse.predict(X_sparse)
+    assert_allclose(y_pred_dense, y_pred_sparse)
+
+    assert_allclose(model_dense[1].intercept_, model_sparse[1].intercept_)
```

Comment:
At first we were reapplying the offset on the intercept but it seems that we always do that:

https://github.com/scikit-learn/scikit-learn/blob/27cfe14c25087e56052657ee50fbc7a7d87645f5/sklearn/linear_model/_base.py#L243-L245

Then the equivalence between the intercept should strict equality. Does it seem correct to you @agramfort?